### PR TITLE
Simplify deposits UI - Add billing schedule section heading

### LINKED ISSUE #5699: Simplify deposits UI - Add note about deposit schedule
### Description

This issue is part of the larger Epic to add the Payments Overview Deposits card (https://github.com/Automattic/woocommerce-payments/issues/5686).

As part of this issue, the card should altered to include the following acceptance criteria: 

### Acceptance criteria 

* "**Deposit history**" heading
* "Your deposits are dispatched **automatically every `{frequency}`**" note. 
* Note should support all valid deposit schedule states, including [temporarily disabled aka deposits suspended](https://woocommerce.com/document/woocommerce-payments/deposits/why-deposits-suspended/).

image

### Designs

P2: paJDYF-7fi-p2

image

### Additional context

Project: pdjTHR-2i5-p2

### PR DESCRIPTION
**Pull request by james-allan**
Fixes #5699 

#### Changes proposed in this Pull Request

This PR adds the section heading before the deposits history table. This section includes details about the current deposit schedule:

### Daily

Screenshot 2023-03-29 at 11 26 00 am

### Weekly

| Tuesday        | Friday           |
| ------------- |:-------------:|
| Screenshot 2023-03-29 at 11 27 23 am      | Screenshot 2023-03-29 at 11 27 57 am | 

### Monthly

| Date (6th)        | Last day           |
| ------------- |:-------------:|
| Screenshot 2023-03-29 at 11 28 19 am      | Screenshot 2023-03-29 at 11 29 32 am | 

### Suspended

Screenshot 2023-03-30 at 5 07 42 pm

#### Testing instructions

1. Set the wp option `_wcpay_feature_simplify_deposits_ui` to `1` to enable the feature flag.
2. Go to the **Payments > Overview** page
3. Note the Deposit History section heading and deposit schedule.
4. Click the link to change your deposit schedule in the footer. 
5. Change the schedule to any combination and confirm the deposit schedule updates in the deposit history section. 

To test the blocked status

> **Note**
> Changes introduced in https://github.com/Automattic/woocommerce-payments/pull/5864 are necessary to prevent an error if you use the Server approach below. 

1. Disable your WC Payments accounts. To do this there are 2 options.
    1. On client. Change the `depositStatus` in [`get_account_status_data()](https://github.com/automattic/woocommerce-payments/blob/issue/5699-display-billing-schedule/includes/class-wc-payments-account.php#L220)` so `'blocked'` is returned. 
    2. On Server change the `is_deposits_blocked()` function to always return `true`.
2. Refresh the **Payments > Overview** page and confirm the deposit schedule section now reads as: "Your deposits are temporarily suspended ([learn more](https://woocommerce.com/document/payments/faq/deposits-suspended/))"

-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `5891` or branch name `issue/5699-display-billing-schedule` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=issue/5699-display-billing-schedule) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 2f2ff15794c7127d5721d9a576607756ab3dd310
- Build time: 2023-04-06 00:31:59 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by james-allan** [comment]
Hey @elizaan36 the designs included a deposit schedule example for weekly with the anchor set to "Friday". As part of this PR I took the liberty to build on the provided format to fill out the rest. Could you provide feedback on the following please?

- Your deposits are dispatched **automatically every Friday** -- _(provided in the designs)_
- Your deposits are dispatched **automatically every day**
- Your deposits are dispatched **automatically on the 10th of every month**
- Your deposits are dispatched **automatically on the last day of every month**
- Your deposits are **temporarily suspended** ([learn more](https://woocommerce.com/document/payments/faq/deposits-suspended/)) -- _(provided [here](https://github.com/Automattic/woocommerce-payments/issues/5699#issuecomment-1478572855))_

I've tried keep them all following the same format ie (**every {period}**) but wanted to get a second opinion because some of them are a bit wordy. I've included screenshots of each in the PR description above.

**Comment by elizaan36** [comment]
All looks good to me except the suspended account.
image

According to [this comment](https://github.com/Automattic/woocommerce-payments/issues/5699#issuecomment-1487938257), I believe the estimated deposit wouldn't appear.

Also "Learn more" has capitalized L and doesn't need parentheses.

**Comment by james-allan** [comment]
Thanks Elizabeth

> According to https://github.com/Automattic/woocommerce-payments/issues/5699#issuecomment-1487938257, I believe the estimated deposit wouldn't appear.

Yep the screenshot isn't accurate. I had to hack it which got me to the false state. We'll need to clarify what the next deposit table should look like in this state, but I'll ask that on p2 for record keeping reasons.

**Comment by james-allan** [comment]
> We'll need to clarify what the next deposit table should look like in this state, but I'll ask that on p2 for record keeping reasons.

You've answered my questions in this comment https://github.com/Automattic/woocommerce-payments/issues/5699

I've updated this PR in line with suggested changes. WC Payment accounts with suspended deposits will now be shown a warning notice. 

Screenshot 2023-03-30 at 5 16 01 pm

Note we haven't implemented the deposit history table yet so the screenshot above looks a little stark.

File: client/components/deposits-overview/next-deposit.tsx [hunk 1]
```diff
@@ -40,23 +40,6 @@ const NextDepositDetails: React.FC< NextDepositProps > = ( {
...
-					/>
-				</span>
-
-				<span className="wcpay-deposits-overview__heading__description">
-					<Loadable
-						isLoading={ isLoading }
-						value={ strings.nextDeposit.description }
-					/>
-				</span>
-			</div>
```
Comment (james-allan) [inline client/components/deposits-overview/next-deposit.tsx:66]: I moved this section of the **`<NextDepositDetails>`** component so it could be reused for the **Deposit History** title and description.

File: client/components/banner-notice/index.tsx [hunk 2]
```diff
@@ -80,7 +80,7 @@ function BannerNotice( props: BannerNoticeProps ): JSX.Element {
 
 	return (
 		<Notice { ...noticeProps }>
-			<Flex align="flex-start" justify="flex-start">
```
Comment (james-allan) [inline client/components/banner-notice/index.tsx:83]: This change and the change below were originally submitted to https://github.com/Automattic/woocommerce-payments/pull/5839 but given that PR will be on-hold for some time, I cherry picked the commit over to this so the notice looks correct.

File: client/components/account-balances/test/index.test.tsx [hunk 3]
```diff
@@ -33,6 +33,7 @@ const mockAccount: AccountOverview.Account = {
 		delay_days: 0,
 		interval: 'weekly',
 		weekly_anchor: 'Monday',
+		monthly_anchor: 1,
```
Comment (james-allan) [inline client/components/account-balances/test/index.test.tsx:36]: In this PR I updated the `AccountOverview.Account` declaration to include the `monthly_anchor`. This change to the test is just to reflect that now required prop.

File: client/components/deposits-overview/style.scss [hunk 4]
```diff
@@ -13,9 +13,13 @@
 		}
 
 		&__description {
-			font-size: 13px;
```
Comment (james-allan) [inline client/components/deposits-overview/style.scss:16]: I removed the `font-size` because it wasn't necessary - the default font size is 13px.

File: client/components/deposits-overview/utils.tsx [hunk 5]
```diff
@@ -0,0 +1,131 @@
+/**
```
Comment (james-allan) [inline client/components/deposits-overview/utils.tsx]: > **Note**
> To be able to use the interpolateComponents function and be able to pass it html elements, I needed to change this file from a `.ts` to a `.tsx` file. So while the file looks like a whole new file in the changeset, I only added 2 new functions: 

`getDepositScheduleDescription()`
`areDepositsBlocked()`

File: client/components/deposits-overview/utils.tsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+ * Fetches the description for the deposit schedule from the account object.
+ *
+ * @param {AccountOverview.Account} account The account object containing information about the deposit schedule.
+ * @return {JSX.Element} The description for the deposit schedule.
+ */
+export const getDepositScheduleDescription = (
+	account: AccountOverview.Account
+): JSX.Element => {
+	const schedule = account.deposits_schedule;
```
Comment (haszari) [inline client/components/deposits-overview/utils.tsx]: Regarding naming (continuing discussion from slack): this could be a regular function component. Also, it looks like it only needs schedule, not the whole account. You might be able to simplify props more by passing only what's needed from `schedule`. Maybe it's more typescript style to pass the smallest "struct" needed, e.g. `schedule`?

Something like: 

```jsx
// Renders a deposit schedule as a descriptive rich-text sentence for display in UI. 
function DepositScheduleDescription( { schedule } ) {
  return (…); 
}

<DepositScheduleDescription schedule={schedule} />
```

File: client/components/deposits-overview/utils.tsx [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+					capitalize: false,
+				} )
+			);
+			break;
+	}
+
+	return interpolateComponents( {
+		mixedString: description,
+		components: {
+			strong: <strong />,
```
Comment (haszari) [inline client/components/deposits-overview/utils.tsx]: I see the string containing this `strong` placeholder is out-of-line, the strings are off in another file. That seems unnecessarily complicated/decoupled to me, I'd say idiomatic react style has the strings and markup needed to render inline in the component function (think template). Then it's easy to reason about and iterate on (self contained). For example, if (in future) we change how we render the schedule string, or add a link to change settings, the changes would be localised to this component. 

I realise we've factored out the help link URLs for easy review / update, but I'm not clear on the rationale for moving regular strings out of components.

Suggestion - move the strings inline so this function is self-contained and reads like a template for `DepositScheduleDescription` component. You could even move to `DepositScheduleDescription` file, or as a sub-component in overview card.

Comment (haszari) [reply client/components/deposits-overview/utils.tsx]: … if you do make this a component, and implement it in its own file or closer to use, then the utils file doesn't need to be tsx, just regular TypeScript.

File: changelog/issue-5699-display-billing-schedule [hunk 8]
```diff
@@ -0,0 +1,5 @@
+Significance: patch
+Type: dev
+Comment: NA. This element will be part of a larger changeset. This PR doesn't need a changelog entry
```
Comment (haszari) [inline changelog/issue-5699-display-billing-schedule:3]: We've gotta get a label or something that allows us to cleanly silence changelog bot!

**Review by haszari — COMMENTED** [review]
Added some specific comments about `getDepositScheduleDescription` (aka `DepositScheduleDescription`). Note I haven't done a full review or test :)

File: changelog/issue-5699-display-billing-schedule (hunk 8 above)
Comment (james-allan) [reply changelog/issue-5699-display-billing-schedule:3]: I've added a retro item. :)

File: client/components/deposits-overview/utils.tsx (hunk 7 above)
Comment (james-allan) [reply client/components/deposits-overview/utils.tsx]: Thanks for the advice @haszari. I've had a crack at what I think you were getting at in [603e10f](https://github.com/Automattic/woocommerce-payments/pull/5891/commits/603e10fbc67a577ea1cb935ca9dd6beb41912991). 

There are now separate components for the billing schedule and suspended notice. Both components only require what they need. ie `depositsSchedule` and nothing respectively. 

I've also been able to return the utils file back to a `.ts` file. :)

Comment (haszari) [reply client/components/deposits-overview/utils.tsx]: Nice! 🙌

File: client/globals.d.ts [hunk 9]
```diff
@@ -8,6 +8,7 @@ declare const wcpaySettings: {
 	connectUrl: string;
 	isSubscriptionsActive: boolean;
 	featureFlags: {
+		customDepositSchedules?: boolean;
```
Comment (Jinksi) [inline client/globals.d.ts]: Looks like this line was inadvertently added while merging `develop` after #5008.

File: client/components/deposits-overview/utils.ts [hunk 10]
```diff
@@ -1,7 +1,16 @@
...
+import React from 'react';
+import { sprintf } from '@wordpress/i18n';
+import moment from 'moment';
+import interpolateComponents from '@automattic/interpolate-components';
+
 /**
  * Internal dependencies
  */
 import { formatCurrency } from 'utils/currency';
+import strings from './strings';
```
Comment (Jinksi) [inline client/components/deposits-overview/utils.ts]: Nit: There are some unused imports in this file.

File: client/globals.d.ts (hunk 9 above)
Comment (james-allan) [reply client/globals.d.ts]: Yep this was removed as part of this commit: https://github.com/Automattic/woocommerce-payments/pull/5891/commits/3d7768e6ca71ae9492a885d627866e715d0e9453 when i updated to the latest `develop` after #5008

File: client/components/deposits-overview/utils.ts (hunk 10 above)
Comment (james-allan) [reply client/components/deposits-overview/utils.ts]: Oh thanks. I've fixed that in [42b629e](https://github.com/Automattic/woocommerce-payments/pull/5891/commits/42b629e7a729ae3c18bac097c0e9761917c22d9c)

**Review by mattallan — APPROVED** [review]
I tested the following deposit schedules:
 - [x] Daily
 - [x] Every week on Monday|Friday day
 - [x] Every month
   - [x] 31st of every month (automatically on the last day of every month)
   - [x] 1st of every month

Confirmed all cases are displaying correctly on the new deposits overview component. Marking as ready to merge! 💯
